File: frameworks/python/django-asgi/Implementation/views/template.py
from django.conf import settings
from django.shortcuts import render
from django.views.decorators.http import require_GET

# template: the payload rendered by Django's own template language, through the TEMPLATES setting.
P = settings.PAYLOADS


# rb:handler template.small
@require_GET
async def small(request):
    return render(request, "items.html", P.small)


# rb:handler template.medium
@require_GET
async def medium(request):
    return render(request, "items.html", P.medium)

File: frameworks/go/chi/Implementation/template.go
package implementation

import (
	"embed"
	"html/template"
	"net/http"

	"github.com/go-chi/chi/v5"
)

// rb:wiring template.*

//go:embed views/items.tmpl
var views embed.FS

// templateRoutes render the payload with html/template. chi has no view layer, and render
// writes only a string it is handed as HTML, so the handler executes the template into the
// response. The template is compiled into the binary and parsed once.
func templateRoutes(r chi.Router, p *Payloads) {
	// rb:wiring template.*
	page := template.Must(template.ParseFS(views, "views/items.tmpl"))
	rendered := func(payload *Payload) http.HandlerFunc {
		return func(w http.ResponseWriter, r *http.Request) {
			w.Header().Set("Content-Type", "text/html; charset=utf-8")
			_ = page.Execute(w, payload)
		}
	}
	// rb:end

	r.Get("/template/small", rendered(&p.Small))

	r.Get("/template/medium", rendered(&p.Medium))
}
